### PR TITLE
[minui] Close open file descriptors upon surface free. Contributes to JB#46010

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Comment by rainemak** [comment]
Thought first upgrading minui to newer fbdev based version (need to disable some Android related bits) from our nile adaptation. Have that branch still and I think that we should aim to update but that's another task and for later time.

**Comment by rainemak** [comment]
Found out that all file descriptors of open_png were leaked. This fixes that issue and now I happily allocate and destroy things in a loop.

@tomin1 @spiiroin @denexter review ?

File: resources.c [hunk 1]
```diff
@@ -289,12 +302,15 @@ int res_create_multi_display_surface(const char* name, int* frames, gr_surface**
         result = -8;
         goto exit;
     }
+
+    (*surface)->_png_fd = fd;
```
Comment (Tomin1) [inline resources.c]: Is this line intended to be here? It looks like the first item in surface array gets _png_fd assigned and then few lines lower it is assigned to NULL without ever touching the value of _png_fd. Should this be just after the loop?

**Review by Tomin1 — COMMENTED** [review]
Just this one thing bothering me here.

File: resources.c (hunk 1 above)
Comment (adenexter) [reply resources.c]: I'm not following the need to cache the file object in the surface either, the file's not going to be read from after res_create_multi_display_surface or similar functions return right? So closing in exit: irrespective of the result should be enough?

Comment (rainemak) [reply resources.c]: Likewise... Initially free'ed too early and some how ended up writing like it was. What a stupid call that was :)

Fixed.

**Comment by rainemak** [comment]
Now all png loading return paths should be covered.

**Comment by adenexter** [comment]
LGTM

**Pull request merged**